backend/src/users/views.ts

```diff
@@ -18,43 +18,50 @@ userRouter.post(
   NoAuth as RequestHandler,
   async (req: Request, res: Response) => {
     // #swagger.tags = ['Users']
+    let user;
     const { password, ...rest } = req.body;
     try {
-      const user = await userController.createUser(
+      // Create user in local database
+      user = await userController.createUser(
         rest,
         rest.profile,
         rest.preferences,
         rest.permissions
       );
-      if (user && process.env.NODE_ENV !== "test") {
-        try {
-          const fbUser = await firebase.auth().createUser({
-            uid: user.id,
-            email: rest.email,
-            password: password,
+
+      // If local user doesn't exist, throw error
+      if (!user) {
+        throw Error("Failed to create local user");
+      }
+
+      // If test environment, return user without firebase auth
+      else if (process.env.NODE_ENV === "test") {
+        return res.status(201).send({ success: true, data: user });
+      }
```

Comment:
Old changes were breaking the test cases because nothing was being returned if the env variable was set to `test`. Added the return back and cleaned up the code a bit.

Reply:
Good catch!